File: tournaments/countNeighbouringPairs/countNeighbouringPairs.cs
int countNeighbouringPairs(string inputString) {
    var result = 0;

    for (int i = 0; inputString.Length > i + 1; ++i) {
        if (inputString[i] == inputString[i + 1]) {
            ++result;
        }
    }

    return result;
}

File: tournaments/countNeighbouringPairs/countNeighbouringPairs.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub countNeighbouringPairs {
    my ($inputString) = @_;

    my $result = 0;
    my @store = split '', $inputString;

    for (my $i = 0; @store > $i + 1; ++$i) {
        if ($store[$i] eq $store[$i + 1]) {
            ++$result;
        }
    }

    return $result;
}
